### PR TITLE
New document containing cluster information

### PR DESCRIPTION
**Pull request by mkarg75**
### Description

This PR adds the gen_metadata function in common.sh which will collect various important cluster information and send them in a separate document to ElasticSearch. 
The function has 3 arguments: The benchmark name, the start date and the end date, so that a call to it could be

```
run_workload() {
  log "Deploying benchmark"
  START_DATE=$(date +%s)
  local TMPCR=$(mktemp)
  envsubst < $1 > ${TMPCR}
  run_benchmark ${TMPCR} ${TEST_TIMEOUT}
  local rc=$?
  END_DATE=$(date +%s)
  log "Generating metadata"
  gen_metadata $WORKLOAD $START_DATE $END_DATE
  if [[ ${TEST_CLEANUP} == "true" ]]; then
    log "Cleaning up benchmark"
    kubectl delete -f ${TMPCR}
  fi
  return ${rc}
}
```

### CODE REVIEW

File: utils/common.sh [hunk 1]
```diff
@@ -106,3 +106,83 @@ gen_spreadsheet() {
...
+
+gen_metadata() {
+        BENCHMARK=$1
+        START_DATE=$2
+        END_DATE=$3
+
+        # construct all the required information
+        local VERSION_INFO=$(oc version -o json)
+        local INFRA_INFO=$(oc get infrastructure.config.openshift.io cluster -o json)
+        local PLATFORM=$(echo ${INFRA_INFO} | jq -r .spec.platformSpec.type)
```
Comment (morenod) [inline utils/common.sh:128]: As I have mentioned on https://github.com/cloud-bulldozer/e2e-benchmarking/pull/332#pullrequestreview-902739288, platform type is not enough to differentiate AWS clusters from ROSA ones.

We need to check the tags included on platformStatus:

```
       "platformStatus": {
            "aws": {
                "region": "us-west-2",
                "resourceTags": [
                    {
                        "key": "red-hat-clustertype",
                        "value": "rosa"
                    },
                    {
                        "key": "red-hat-managed",
                        "value": "true"
                    }
                ]
            },
            "type": "AWS"
        }
    }
```

**Review by morenod — CHANGES_REQUESTED** [review]
added some info to the platform parameter

File: workloads/cyclictest/library.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```
Comment (rsevilla87) [inline workloads/cyclictest/library.sh:1]: Do we need this file here?

Comment (mkarg75) [reply workloads/cyclictest/library.sh]: removed

File: utils/common.sh (hunk 1 above)
Comment (mkarg75) [reply utils/common.sh:128]: Checking if resourceTags[0].value is rosa now and adding it as CLUSTERTYPE to the document, let me know if that's enough

Comment (morenod) [reply utils/common.sh:128]: yep, now it is perfect

**Review by morenod — APPROVED** [review]
LGTM

**Review by rsevilla87 — APPROVED** [review]
LGTM!

**Pull request merged**